packages/contracts/src/utils/JwtRegistry.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "@zk-email/contracts/DKIMRegistry.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { strings } from "solidity-stringutils/src/strings.sol";
+
+/// @title JWT Registry
+/// @notice TODO
+/// @dev TODO
+contract JwtRegistry is IDKIMRegistry, Ownable {
+
+    using strings for *;
+
+    DKIMRegistry public dkimRegistry;
+
+    // Check if azp is registered
+    mapping(string => bool) public whitelistedClients;
+
+    constructor(address _owner) Ownable(_owner) {
+        dkimRegistry = new DKIMRegistry(address(this));
+    }
+
+    /// @notice Checks if a public key hash is valid and not revoked for a given kis and iss.
+    /// @param domainName The domain name contains kis, iss and azp fields.
+    /// @param publicKeyHash The public key hash to validate.
+    /// @return bool Returns true if the public key hash is valid and not revoked, false otherwise.
+    function isDKIMPublicKeyHashValid(
+        string memory domainName,
+        bytes32 publicKeyHash
+    ) public view returns (bool) {
+        string[] memory parts = this.stringToArray(domainName);
+        string memory kidAndIss = string(abi.encode(parts[0], "|", parts[1]));
+        return dkimRegistry.isDKIMPublicKeyHashValid(kidAndIss, publicKeyHash) 
+          && whitelistedClients[parts[2]];
+    }
```

Comment:
This is not a DKIM public key, it's a JWT verification public key

Reply:
@shreyas-londhe function names, etc. have not been changed to make the interface match the previous DKIMRegistry. @SoraSuegami